components/brave_wallet/renderer/js_ethereum_provider.cc

```diff
@@ -135,6 +128,15 @@ void JSEthereumProvider::DidDispatchDOMContentLoadedEvent() {
   ConnectEvent();
 }
 
+void JSEthereumProvider::DidFinishLoad() {
```

Comment:
pls add a comment to this function explaining why we bind/announce things here and not in the prevous place, with a link to the issue.

Also can you check similar renderer injections in other `RenderFrameObserver`s in our code? Is it possible a similar problem exists in other code?

Reply:
Added comment 
Based on quick research this bug needs `DidClearWindowObject`, `gin::CreateFunctionTemplate`, `web_frame->CallFunctionEvenIfScriptDisabled` set of calls to potentially crash. And it seems it doesn't appear anywhere else

Reply:
thanks @supermassive